src/Twilio/Clients/BearerToken/TwilioBearerTokenRestClient.cs

```diff
@@ -0,0 +1,317 @@
+        /// <param name="edge">edge to make requests for</param>
+        public TwilioBearerTokenRestClient(
+            TokenManager tokenManager,
+            string region = null,
+            BearerTokenHttpClient httpClient = null,
+            string edge = null
+        )
+        {
+            _tokenManager = tokenManager;
+
+            HttpClient = httpClient ?? DefaultClient();
+
+            Region = region;
+            Edge = edge;
+        }
+
+        /// <summary>
+        /// Make a request to the Twilio API
+        /// </summary>
+        ///
+        /// <param name="request">request to make</param>
+        /// <returns>response of the request</returns>
+        public Response Request(BearerTokenRequest request)
+        {
+
+            if (_accessToken == null ){
+            //|| isTokenExpired(_accessToken)) {
+                lock (lockObject){
+                    if (_accessToken == null){
+                     //|| isTokenExpired(_accessToken)) {
+                        _accessToken = _tokenManager.fetchAccessToken();
+                    }
+                }
+            }
+            request.SetAuth(_accessToken);
+
+            if (LogLevel == "debug")
+                LogRequest(request);
+
+            if (Region != null)
+                request.Region = Region;
+
+            if (Edge != null)
+                request.Edge = Edge;
+
+            if (UserAgentExtensions != null)
+                request.UserAgentExtensions = UserAgentExtensions;
+
+            Response response;
+            try
+            {
+                response = HttpClient.MakeRequest(request);
+                if (LogLevel == "debug")
+                {
+                    Console.WriteLine("response.status: " + response.StatusCode);
+                    Console.WriteLine("response.headers: " + response.Headers);
+                }
+            }
+            catch (Exception clientException)
+            {
+                throw new ApiConnectionException(
+                    "Connection Error: " + request.Method + request.ConstructUrl(),
+                    clientException
+                );
+            }
+            return ProcessResponse(response);
+        }
+
+        /// <summary>
+        /// To check if token is expired or not
+        /// </summary>
+        ///
+        /// <param name="token">token to validate</param>
+        /// <returns>True if token is not expired, false otherwise</returns>
+//        public bool isTokenExpired(String token) {
+//
+//            var tokenHandler = new JwtSecurityTokenHandler();
+//            var jwtToken = tokenHandler.ReadJwtToken(token);
+//            var expirationTime = jwtToken.Payload.Exp.HasValue
+//                        ? DateTimeOffset.FromUnixTimeSeconds(jwtToken.Payload.Exp.Value)
+//                        : DateTimeOffset.MinValue;
+//           return expirationTime <= DateTimeOffset.UtcNow;
```

Comment:
Make sure that this gives same result, irrespective of system time.
Example: If run on system with IST time zone and if run on different time zone, result should not vary.

Reply:
this function implementation is bit tricky, tracked as a different ticket in next sprint